### DISCUSSION TITLE
How to solve the following error, "Check for incorrect argument data type or missing argument in call to function 'sdpvar'."

### DISCUSSION THREAD

**Discussion by ToshihideH** [yalmip/YALMIP#1366] [Problems using YALMIP] [answered]
Hello, sir. When I implemented the following code, I got the error in title.
`
%Filename = sf_H2IP.m

%model Setup
for i=1:3
    for j=1:2
        [A(:,:,i,j),B(:,:,i,j)]=pend_model(0.1*(i+2),1+0.9*j);
    end
end

Bw = [0 pi/36 0 0]' ;
Cz = [ eye(4) ; zeros(1,4) ]; 
Dzu = [ 0 0 0 0 1 ]' ;

%LMI Setup
X = sdpvar(4,4,'symmetric');
Y = sdpvar(1,4,'full');
gamma = sdpvar(1,1,'symmetric');

%LMI Descripstion
LMI = [];
for i=1:3
    for j=1:2
        M = [He(A(:,:,i,j)*X + B(:,:,i,j)*Y) (Cz*X+Dzu*Y)'
            Cz*X+Dzu*Y -gamma*eye(5) ];
        LMI = [ LMI, M<=0 ];
    end
end

LMI = [ LMI, [ X Bw ; Bw' 1 ] >=0, [ X Y' ; Y 1.2^2 ] >=0 ];

%SDP Solution
sol = solvesdp(LMI,gamma);
X = double(X);
Y = double(Y);
gamma = double(gamma);
K = Y/X;
`
And, the error is the following.

Error in sf_H2IP (line 15)
X = sdpvar(4,4,'symmetric');

Where do I have to correct in the code?

**Comment by johanlofberg** [comment, marked as answer]
Your function sdpvar cannot be a correctly installed version of YALMIPs sdpvar since it crashes in that very basic command

Hence, which -all sdpvar will probably reveal that sdpvar is something else than YALMIPs sdpvar, so a correct clean (re)installation is likely the way to go

**Reply by ToshihideH** [reply]
Reinstalling YALMIP solved the problem. Thank you very much!!
